### DISCUSSION TITLE
Using React Query `v4` together with React `v17`

### DISCUSSION THREAD

**Discussion by bennettdams** [TanStack/query#4199] [Q&A] [answered]
I know that React Query `v4` now comes with first-class support for React `v18`, it is also mentioned in the migration guide.
But how does `v4` work with React `v17`? Isn't `v4` using `useSyncExternalStore`, which only exists for React `v18`?

The `v4` still has React `v17` as a [valid peer dependency](https://github.com/TanStack/query/blob/main/packages/react-query/package.json#L54) - are there now two "implementations" in `v4`, one with and one without `useSyncExternalStore`?
What are the differences of using `v4` with React `v17` vs. `v18`? Also keeping in mind Suspense etc.

**Comment by TkDodo** [comment, marked as answer]
We are using the `use-sync-external-store` shim provided by react. It uses the new hook if available and falls back to an implementation with useEffect for v17. other libs like redux do the same afaik.

https://github.com/TanStack/query/blob/39328662c40af700ce3c3c9ddcbfdc605a6887b2/packages/react-query/package.json#L51

We run our tests both against v17 and v18, so both are fully supported.
